components/org.wso2.carbon.identity.api.server.notification.sender/org.wso2.carbon.identity.api.server.notification.sender.v2/src/main/java/org/wso2/carbon/identity/api/server/notification/sender/v2/core/NotificationSenderManagementService.java

```diff
@@ -0,0 +1,493 @@
+        emailSender.setName(dto.getName());
+        emailSender.setFromAddress(dto.getFromAddress());
+        emailSender.setSmtpPort(dto.getSmtpPort());
+        emailSender.setSmtpServerHost(dto.getSmtpServerHost());
+        emailSender.setAuthType(dto.getAuthType());
+        List<Properties> properties = new ArrayList<>();
+
+        // Exclude credentials from the response.
+        Set<String> excludedKeys = new HashSet<>(Arrays.asList(PASSWORD, USERNAME, CLIENT_SECRET, CLIENT_ID));
+
+        if (dto.getProperties() == null) {
+            return emailSender;
+        }
+        dto.getProperties().forEach((key, value) -> {
+            if (StringUtils.isBlank(key) || StringUtils.isBlank(value) || excludedKeys.contains(key)) {
+                return;
+            }
+            Properties prop = new Properties();
+            prop.setKey(key);
+            prop.setValue(value);
+            properties.add(prop);
+        });
+        emailSender.setProperties(properties);
+        return emailSender;
+    }
+
+    private SMSSenderDTO buildSMSSenderDTO(SMSSenderAdd smsSenderAdd) {
+
+        SMSSenderDTO dto = new SMSSenderDTO();
+        dto.setName(smsSenderAdd.getName());
+        dto.setProvider(smsSenderAdd.getProvider());
+        dto.setProviderURL(smsSenderAdd.getProviderURL());
+        dto.setKey(smsSenderAdd.getKey());
+        dto.setSecret(smsSenderAdd.getSecret());
+        dto.setSender(smsSenderAdd.getSender());
+        dto.setContentType(smsSenderAdd.getContentType().toString());
+        List<Properties> properties = smsSenderAdd.getProperties();
+        if (properties != null) {
+            properties.forEach((prop) -> dto.getProperties().put(prop.getKey(), prop.getValue()));
+        }
+        return dto;
+    }
+
+
+    private SMSSenderDTO buildSMSSenderDTO(String senderName, SMSSenderUpdateRequest smsSenderUpdateRequest) {
+
+        SMSSenderDTO dto = new SMSSenderDTO();
+        dto.setName(senderName);
+        dto.setProvider(smsSenderUpdateRequest.getProvider());
+        dto.setProviderURL(smsSenderUpdateRequest.getProviderURL());
+        dto.setKey(smsSenderUpdateRequest.getKey());
+        dto.setSecret(smsSenderUpdateRequest.getSecret());
+        dto.setSender(smsSenderUpdateRequest.getSender());
+        dto.setContentType(smsSenderUpdateRequest.getContentType().toString());
+        List<Properties> properties = smsSenderUpdateRequest.getProperties();
+        if (properties != null) {
+            properties.forEach((prop) -> dto.getProperties().put(prop.getKey(), prop.getValue()));
+        }
+        return dto;
+    }
+
+    private APIError handleException(NotificationSenderManagementException e) {
+
+        if (e instanceof NotificationSenderManagementClientException) {
+            throw buildClientError(e);
+        }
+        throw buildServerError(e);
+    }
+
+    private SMSSender buildSMSSenderFromDTO(SMSSenderDTO dto) {
+
+        SMSSender smsSender = new SMSSender();
```

Comment:
Hope these dto is null safe.

Reply:
Not relevant since this PR focuses only on EmailSender improvements